liquidprompt

```diff
@@ -1397,9 +1433,61 @@ _lp_kubernetes_context() {
 _lp_kubernetes_context_color() {
     _lp_kubernetes_context || return "$?"
 
-    lp_kubernetes_context_color="[${LP_MARK_KUBECONTEXT}${LP_COLOR_KUBECONTEXT}${lp_kubernetes_context}${lp_kubernetes_namespace+:}${lp_kubernetes_namespace-}${NO_COL}]"
+    lp_kubernetes_context_color="${LP_MARK_KUBECONTEXT}${LP_COLOR_KUBECONTEXT}${lp_kubernetes_context}${lp_kubernetes_namespace+:}${lp_kubernetes_namespace-}${NO_COL}"
 }
 
+_lp_cmake() {
+    (( LP_ENABLE_CMAKE )) || return 2
+    [[ -f CMakeCache.txt ]] || return 1
+
+    local cmake_compiler
+    IFS="=" read -r _ cmake_compiler <<<"$(grep COMPILER:FILEPATH CMakeCache.txt)"
+    lp_cmake_compiler="${cmake_compiler##*/}" # Only the part after the last slash.
+
+    local cmake_generator
+    IFS="=" read -r _ cmake_generator <<<"$(grep CMAKE_GENERATOR:INTERNAL CMakeCache.txt)"
+    # Shorten: Makefiles -> Make, and Visual Studio -> VS.
+    cmake_generator="${cmake_generator/%Makefiles/Make}"
+    cmake_generator="${cmake_generator/#Visual Studio/VS}"
+    # Remove all white spaces.
+    lp_cmake_generator="${cmake_generator// /}"
+
+    IFS="=" read -r _ lp_cmake_buildtype <<<"$(grep CMAKE_BUILD_TYPE:STRING CMakeCache.txt)"
+
+    [[ -n "${lp_cmake_compiler}${lp_cmake_generator}${lp_cmake_buildtype}" ]] || return 3
+}
+
+_lp_cmake_color() {
+    (( LP_ENABLE_COLOR )) || return 2
+
+    if _lp_cmake ; then
+        local lp_hash_color
+
+        _lp_hash_color "$lp_cmake_compiler"
+        lp_cmake_compiler="$lp_hash_color"
+
+        _lp_hash_color "$lp_cmake_generator"
+        lp_cmake_generator="$lp_hash_color"
+
+        if [[ "$lp_cmake_buildtype" == "Release" ]] ; then
+            lp_cmake_buildtype="${LP_COLOR_CMAKE_RELEASE}${lp_cmake_buildtype}${NO_COL}"
+        elif [[ "$lp_cmake_buildtype" == "RelWithDebInfo" ]] ; then
+            lp_cmake_buildtype="${LP_COLOR_CMAKE_RWDI}${lp_cmake_buildtype}${NO_COL}"
+        elif [[ "$lp_cmake_buildtype" == "Debug" ]] ; then
+            lp_cmake_buildtype="${LP_COLOR_CMAKE_DEBUG}${lp_cmake_buildtype}${NO_COL}"
+        else
+            _lp_hash_color "$lp_cmake_buildtype"
+            lp_cmake_buildtype="$lp_hash_color"
+        fi
+
+        lp_cmake_color="${lp_cmake_compiler}${LP_MARK_CMAKE}${lp_cmake_generator}${LP_MARK_CMAKE}${lp_cmake_buildtype}"
```

Comment:
Why overwrite these variables? I think string construction will be cleaner, and will leave the original variables they way they were:

```suggestion
        _lp_hash_color "$lp_cmake_compiler"
        lp_cmake_color="${lp_hash_color}${LP_MARK_CMAKE}"

        _lp_hash_color "$lp_cmake_generator"
        lp_cmake_color+="${lp_hash_color}${LP_MARK_CMAKE}"

        if [[ "$lp_cmake_buildtype" == "Release" ]] ; then
            lp_cmake_color+="${LP_COLOR_CMAKE_RELEASE}${lp_cmake_buildtype}${NO_COL}"
        elif [[ "$lp_cmake_buildtype" == "RelWithDebInfo" ]] ; then
            lp_cmake_color+="${LP_COLOR_CMAKE_RWDI}${lp_cmake_buildtype}${NO_COL}"
        elif [[ "$lp_cmake_buildtype" == "Debug" ]] ; then
            lp_cmake_color+="${LP_COLOR_CMAKE_DEBUG}${lp_cmake_buildtype}${NO_COL}"
        else
            _lp_hash_color "$lp_cmake_buildtype"
            lp_cmake_color+="${lp_hash_color}${NO_COL}"
        fi
```